lib/active_fedora/associations/id_composite.rb

```diff
@@ -0,0 +1,30 @@
+module ActiveFedora::Associations
+  ##
+  # A composite object for an array of IDs. This abstracts away the fact that an
+  # ID might be either a relative ID or a URI to a resource.
+  class IDComposite
+    attr_reader :ids, :id_translator
+    include Enumerable
+    def initialize(ids, id_translator)
+      @ids = ids
+      @id_translator = id_translator
+    end
+
+    # @return [Array<relative_id>]
+    def each
+      ids.each do |id|
+        yield convert(id)
+      end
+    end
+
+    private
+
+    def convert(id)
```

Comment:
This may be worth extracting to a value object - an object which can return an ID whether it's given an ID or URI is probably useful outside of the array context.